routes/api/gamestest.js

```diff
@@ -165,4 +167,95 @@ router.get("/search/platform/:search", async (req, res) => {
   }
 });
 
+
+//// ---------review------------------------- /////
+
+
+// Get reviews for a specific game
+router.get('/review/:id', auth, async (req, res) => {
+  try {
+    const game = await Game.findOne({ rawg_id: req.params.id }).populate({
+      path: 'reviews',
+      populate: {
+        path: 'author',
+        model: 'User',
+        select: 'name' // Assuming the user's name is stored in 'name' field
+      }
+    });
+    if (!game) {
+      return res.status(404).json({ msg: "Game not found" });
+    }
+    res.json(game.reviews);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server error');
+  }
+});
+
+
+// @route   POST api/games/review/:id
+// @desc    Post a review on a video game
+// @access  Private
+// @route   POST api/games/review/:id
+// @desc    Post a review on a video game
+// @access  Private
+router.post("/review/:id", auth, async (req, res) => {
+  try {
+    const game = await Game.findOne({ rawg_id: req.params.id }).populate("reviews");
+    if (!game) {
+      return res.status(404).json({ msg: "Video game not found" });
+    }
+
+    const newReview = new Review({
+      body: req.body.body,
+      rating: req.body.rating,
+      author: req.user.id,
+    });
+
+    await newReview.save();
+    game.reviews.unshift(newReview._id);
+    await game.save();
+    res.json(game.reviews);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send("Server error");
+  }
+});
+
+
+
+// @route   DELETE api/games/review/:id/:review_id
+// @desc    Delete a review on a video game
+// @access  Private
+// Correcting the route to use the appropriate identifier
+router.delete("/review/:id/:review_id", auth, async (req, res) => {
+  try {
+    const game = await Game.findOne({ rawg_id: req.params.id });
+    if (!game) {
+      return res.status(404).json({ msg: "Video game not found" });
+    }
+    const review = await Review.findById(req.params.review_id);
+    if (!review) {
+      return res.status(404).json({ msg: "Review does not exist" });
+    }
+    const user = await User.findById(req.user.id);
+    if (review.author.toString() !== req.user.id && user.role !== "admin") {
+      return res.status(401).json({ msg: "User not authorized" });
+    }
+    await review.deleteOne();
```

Comment:
Here you are using detete one and then filtering the games without the deleted review ID. Does node and express have a simpler delete function? That way you don't need to filter through the games.